react/src/RemoteService.js

```diff
@@ -17,15 +17,14 @@ export default class RemoteService {
       url += params.join('&');
 
       const http = new window.XMLHttpRequest();
+      http.timeout = 5000;
       http.open('GET', url, true);
       http.setRequestHeader('Content-type', 'application/json');
       http.onreadystatechange = function onreadystatechange() {
-        if (http.readyState === 4 && http.status === 200) {
```

Comment:
Are you sure that this is correct and will not lead to wrong rejects? For example, when the readyState changes from OPENED to HEADERS_RECEIVED? See: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState